### DISCUSSION TITLE
Adding Themes and Plugins to your ServerlessWP sites

### DISCUSSION THREAD

**Discussion by mitchmac** [mitchmac/ServerlessWP#35] [General]
To add themes and plugins to your ServerlessWP sites, you'll need to place the theme or plugin files into your site's GitHub repository.

Theme files should go in `wp/wp-content/themes` and plugins should go in `wp/wp-content/plugins`.

This is because serverless functions do not reliably keep uploaded files in place between requests.

Check out [this document](https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository) to learn more about adding files to a git repository.

**Comment by alexandrudinca** [comment]
Hi @mitchmac! I tried to add a new plugin in the corresponding folder but this won't show up after redeploy. Is this a bug?

**Reply by k2jama** [reply]
Did you find any solution? I'm having a similar issues with themes. Uploaded a theme, redeployed, and it still isnt recognized... Any help would be really appreciated!

**Reply by ghost** [reply]
hey, still no update on this?

**Reply by ghost** [reply]
hey i did find  a fix, just redeploy and it will show up

**Comment by techcoder01** [comment]
Ok! I have installed the theme and i uploaded the plugins too , but i want to choose a site in the templates , it requires plugins , i have already installed but it say download the plugins which is giving error "Insufficient permissions to install the plugin." Please fix it , OR guide me

**Comment by JinHeyong** [comment]
Very good idea, I hope you can give me a step on how to add a theme, I tried to put the theme into the theme directory, but I couldn't find what I added

**Reply by ghost** [reply]
just redeploy after adding themes or plugins and it'll show up

**Reply by alietverse99** [reply]
I did that but still no success

**Comment by ghost** [comment]
There is no free plan now. How to enter the database when using other
database platforms?



it does still have a free plan , it's just not available in few countries,
try using a VPN with it's server in usa or Japan and you'll be able to
create a free account.
"它仍然有免費計劃，只是在少數國家/地區不可用，嘗試在美國或日本的伺服器上使用 VPN，您將能夠建立免費帳戶。"

**Comment by aumentamas01** [comment]
Los plugins y temas es mejor subirlos desde github app instalada en la PC, luego haces push para actualizar el repositorio en github web. El problema que tengo es que elementor no carga, no me deja editar ninguna pagina con elementor, solo se queda cargando

**Reply by daniel-alejandro-t** [reply]
Elementor requiere estrictamente una base de datos relacional, lo siento.

**Reply by aumentamas01** [reply]
que solucion o como se puede trabajar con los temas?

**Comment by took3000** [comment]
I committed the plugin folder to the repository and deployed it via Vercel, but refreshing the plugins page doesn't show the new plugin. What should I do?
<img width="638" height="592" alt="Snipaste_2026-09-15_11-33-33" src="https://github.com/user-attachments/assets/75a94729-015e-4336-97b6-5daa5e2429eb" />
<img width="285" height="276" alt="Snipaste_2026-09-15_11-33-14" src="https://github.com/user-attachments/assets/23ac2ee4-6658-4bf5-9617-e72b307a4d52" />
